Lib/test/test_imghdr.py

```diff
@@ -77,13 +77,13 @@ def test_file_pos(self):
             self.assertEqual(stream.tell(), pos)
 
     def test_bad_args(self):
-        with self.assertRaises(TypeError):
+        with self.assertRaises(ValueError):
             imghdr.what()
         with self.assertRaises(AttributeError):
             imghdr.what(None)
         with self.assertRaises(TypeError):
             imghdr.what(self.testfile, 1)
-        with self.assertRaises(AttributeError):
+        with self.assertRaises(BytesWarning):
```

Comment:
Given that the signature change should not affect this call, I don't understand the error change.